staking/programs/integrity-pool/src/state/pool.rs

```diff
@@ -312,9 +309,25 @@ impl PoolData {
                 i,
                 eligible_delegation_data.get_reward_ratios()?,
             )?;
+            total_eligible_delegation +=
+                eligible_delegation_data.get_total_eligible_delegation() * (epochs_passed - 1);
             i += 1;
         }
 
+        let (adjusted_y, adjusted_rewards_to_be_distributed) = self.adjust_rewards_if_needed(
+            y,
+            total_eligible_delegation,
+            pool_reward_custody_balance,
+        )?;
+        for epoch in self.last_updated_epoch..current_epoch {
+            let event =
+                self.get_event_mut((self.num_events + epoch - self.last_updated_epoch).try_into()?);
+            event.epoch = epoch;
+            event.y = adjusted_y;
```

Comment:
does this mean that we distribute the remaining tokens in all the epochs that have passed?

Reply:
Yes